internal/scheduler/scheduler_test.go

```diff
@@ -665,7 +665,7 @@ type testSchedulingAlgo struct {
 	shouldError    bool
 }
 
-func (t *testSchedulingAlgo) Schedule(ctx context.Context, txn *jobdb.Txn, jobDb *jobdb.JobDb) ([]*jobdb.Job, error) {
```

Comment:
shouldn't there be a test here that we do indeed create the correct prempted messages?